pkg/storegateway/replication_strategy.go

```diff
@@ -0,0 +1,45 @@
+package storegateway
+
+import (
+	"errors"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/ring"
+)
+
+type ReplicationStrategy struct{}
```

Comment:
This is exported because it will be used the querier too (next PR).

I personally find `StoreGatewayReplicationStrategy` a bit redundant. It's quite common in Go avoiding same package prefix (the package name is `storegateway` so you would externally access it via `storegateway.StoreGatewayReplicationStrategy`). Thoughts?

Reply:
You don't always see package name when working with the type, but also agree that `StoreGatewayReplicationStrategy` is redundant.

What about `BlocksReplicationStrategy`?

Reply:
Makes sense, done.